src/main/scala/sigmastate/serialization/ErgoTreeSerializer.scala

```diff
@@ -111,10 +115,10 @@ class ErgoTreeSerializer {
     r.constantStore = new ConstantStore(constants)
     val previousResolvePlaceholderValue = r.resolvePlaceholdersToConstants
     r.resolvePlaceholdersToConstants = resolvePlaceholdersToConstants
-    val tree = ValueSerializer.deserialize(r)
+    val value = ValueSerializer.deserialize(r)
```

Comment:
@aslesarenko I did not add a type check (is `SigmaProp`) here since the `serializeWithSegregation` us used for `Value.bytes` in [src](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/ac5a03c152e66162a48ebe4a080479565c42b92a/src/main/scala/sigmastate/Values.scala#L58).

Reply:
@greenhat do we need this Value.bytes at all?
I suggest to remove it completely. 
It is used mostly in Specifications and should be replaced by ErgoTree serialization.
The only core usage is inside MonetarySettings where it also should be replaces with ErgoTree serialization. /cc @catena2w @kushti

Reply:
I've removed `Value.bytes` and switch ser/deser methods to `SigmaPropValue` with added type check on deserialization. See [commit](https://github.com/ScorexFoundation/sigmastate-interpreter/pull/459/commits/f72682893e1e2676a6befcc223b28948ba920cce).

Reply:
@greenhat I really mean completely :) 
We don't need any public means to serialize Value directly. 
The only script serialization should be via explicit construction of ErgoTree. 
Otherwise it leads to confusion, how the bytes should be deserialized, and how to interpret result of such deserialization.

Reply:
@aslesarenko Done in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/459/commits/6edd1f4089a2df3ab8b4b4fc25dc8e7cdffa462e